packages/desktop/src/index.ts

```diff
@@ -106,149 +19,98 @@ if (require('electron-squirrel-startup')) {
-      // update the settings & title
-      const settings = readSettings();
-      settings.workspace = path.dirname(file.filePath);
-      writeSettings(settings);
-      mainWindow.setTitle(`Data Story - ${file.filePath}`);
-
-      result.isSuccess = true;
-    }
-
-    return result;
-  } catch(err) {
-    result.data = err;
-    return result;
+  constructor() {
+    this.createWindow();
   }
-});
 
-ipcMain.handle('openDiagram', open);
-
-const createWindow = (): void => {
-  // Create the browser window.
-  mainWindow = new BrowserWindow({
-    height: 1600,
-    width: 1800,
-    webPreferences: {
-      nodeIntegration: false,
-      contextIsolation: true,
-      preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY,
-    },
-  });
+  public getMainWindowActions(): MainWindowActions {
+    return {
+      setTitle: (title) => this.mainWindow.setTitle(title),
+      webContentsSend: (channel, data) => this.mainWindow.webContents.send(channel, data),
+    };
+  }
 
-  // Modify the Content Security Policy
-  mainWindow.webContents.session.webRequest.onHeadersReceived((details, callback) => {
-    callback({
-      responseHeaders: {
-        ...details.responseHeaders,
-        'Content-Security-Policy': [
-          'default-src \'self\' \'unsafe-inline\' data:; ' +
-          'script-src \'self\' \'unsafe-eval\' \'unsafe-inline\' ws://localhost:3100; ' +
-          'connect-src \'self\' ws://localhost:3100'
-        ]
-      }
+  public getWorkspace(): Workspace {
+    return this.workspace;
+  }
+
+  public switchWorkspace(filePath: string): void {
+    // Assuming Workspace is a defined class that takes a filePath in its constructor
+    this.workspace = new Workspace(filePath);
+    this.workspace.initSettingsAndEnv(this.getMainWindowActions());
+  }
+
+  public initWorkspace(): void {
+    const mainWindowActions = this.getMainWindowActions()
+
+    // Assuming DefaultWorkspace is a defined class
+    this.workspace = new DefaultWorkspace();
+    this.workspace.initSettingsAndEnv(mainWindowActions);
+  }
+
+  private createWindow(): void {
+    // Create the browser window.
+    this.mainWindow = new BrowserWindow({
+      height: 1600,
+      width: 1800,
+      webPreferences: {
+        nodeIntegration: false,
+        contextIsolation: true,
+        preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY, // 确保这个变量在类的上下文中是可访问的
+      },
     });
-  });
 
-  // and load the index.html of the app.
-  mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
-  // Open the DevTools.
-  mainWindow.webContents.openDevTools();
+    // Modify the Content Security Policy
+    this.mainWindow.webContents.session.webRequest.onHeadersReceived((details, callback) => {
+      callback({
+        responseHeaders: {
+          ...details.responseHeaders,
+          'Content-Security-Policy': [
+            'default-src \'self\' \'unsafe-inline\' data:; ' +
+            'script-src \'self\' \'unsafe-eval\' \'unsafe-inline\' ws://localhost:3100; ' +
```

Comment:
Lets extract port to a variable

Reply:
Love the idea! 💫  I'm planning to implement the change in the next PR (`obtain a free port automatically in case of a conflict`). What are your thoughts?

Reply:
Perfect!